docs/enhancements/0002_day0-integration.md

```diff
@@ -0,0 +1,285 @@
+kind: ModuleDay0
+spec:
+  osMapping:
+    literal: "Red Hat Enterprise Linux CoreOS 414.92.202312311229-0 (Plow)"
+    containerImage: quay.io/ybettan/rhcos:414.92.202312311229-0
+    build:
+      secrets:
+        - name: build-secret
+      dockerfileConfigMap:
+        name: kmm-kmod-dockerfile
+  machineConfigPoolSelector:
+    node-role.kubernetes.io/worker: ""
+```
+
+The build will trigger if all the following conditions are met
+  * `quay.io/ybettan/rhcos:414.92.202312311229-0` is different than `mc.osImageURL`
+  * `quay.io/ybettan/rhcos:414.92.202312311229-0` doesn't exist
+  * A `build` section exists in the `osMapping`
+
+Once built, KMM will modify the `mc.osImageURL` with the new image.
+
+The `ConfigMap` should be in the following format
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kmm-kmod-dockerfile
+data:
+  dockerfile: |
+
+    ARG DTK_AUTO
+    ARG OS_AUTO
+
+    FROM ${DTK_AUTO} as builder
+
+    ARG KERNEL_VERSION
+    ...
+    RUN KERNEL_SRC_DIR=/lib/modules/${KERNEL_VERSION}/build make all
+
+    FROM ${OS_AUTO}
+
+    ARG KERNEL_VERSION
+
+    COPY --from=builder /usr/src/kernel-module-management/ci/kmm-kmod/kmm_ci_a.ko /opt/lib/modules/${KERNEL_VERSION}/
+    COPY --from=builder /usr/src/kernel-module-management/ci/kmm-kmod/kmm_ci_b.ko /opt/lib/modules/${KERNEL_VERSION}/
+    RUN depmod -b /opt ${KERNEL_VERSION}
+```
+
+`OS_AUTO` will fetch the `rhel-coreos` image from the release-payload of the
+cluster.
+The DTK `ImageStream` on the cluster will contain the DTK image that fits the
+release-payload and not the running OS on the nodes (which may have a different
+kernel), therefore, `DTK_AUTO` should only be used if `OS_AUTO` is used.
+
+### Signing
+
+If a customer wish us to sign his kmods, he should add a `sign` section
+in the `ModuleDay0` as follow
+```
+apiVersion: kmm.sigs.x-k8s.io/v1beta1
+kind: ModuleDay0
+spec:
+  osMapping:
+    literal: "Red Hat Enterprise Linux CoreOS 414.92.202312311229-0 (Plow)"
+    containerImage: quay.io/ybettan/rhcos:414.92.202312311229-0
+    sign:
+      unsignedImage: quay.io/ybettan/rhcos:414.92.202312311229-0-unsigned
+      certSecret:
+        name: kmm-kmod-signing-cert
+      keySecret:
+        name: kmm-kmod-signing-key
+      filesToSign:
+        - /opt/lib/modules/${KERNEL_FULL_VERSION}/kmm_ci_a.ko
+  machineConfigPoolSelector:
+    node-role.kubernetes.io/worker: ""
+```
+
+The sign will trigger if all the following conditions are met
+  * `quay.io/ybettan/rhcos:414.92.202312311229-0` is different than `mc.osImageURL`
+  * `quay.io/ybettan/rhcos:414.92.202312311229-0-unsigned` exist (or a `build` section exists)
+  * A `sign` section exists in the `osMapping`
+
+Once signed and push, KMM will modify the `mc.osImageURL` with the new image.
+
+### Firmware
+
+No much to do here, the user can put his firmware files directly in
+`/var/lib/firmware` on the nodes in the custom container image so the firmware
+feature of KMM is a bit useless here.
```

Comment:
`/var/lib/firmware` is not a default lookup path for the kernel. Users should either:
- put the firmware files in `/lib/firmware`, checking carefully that they don't overwrite anything, or;
- set `firmware_class.path` in their image's `sysctl.conf`.

Reply:
Fixed. https://github.com/rh-ecosystem-edge/kernel-module-management/pull/957/commits/d0612e764912692cfe3c0653212252f407885b6e